internal/testrunner/runners/policy/policy.go

```diff
@@ -184,6 +214,46 @@ func cleanPolicy(policy []byte, entriesToClean []policyEntryFilter) ([]byte, err
 	return yaml.Marshal(policyMap)
 }
 
+// replaceOtelComponentIDs finds OTel Collector component IDs in the policy and replaces them with controlled values.
+// It also replaces references to those IDs in service.extensions and service.pipelines.
+func replaceOtelComponentIDs(policy []byte) []byte {
+	replacementsDone := map[string]string{}
+
+	policy = otelComponentIDsRegexp.ReplaceAllFunc(policy, func(match []byte) []byte {
+		count := 0
+		lines := bytes.Split(match, []byte("\n"))
+		for i, line := range lines {
+			line = bytes.TrimRight(line, "\r\n")
+			stringLine := string(line)
+			if strings.Contains(stringLine, ":") {
+				if uniqueOtelComponentIDReplace.regexp.MatchString(stringLine) {
+					replacement := fmt.Sprintf(uniqueOtelComponentIDReplace.replace, strconv.Itoa(count))
+					count++
+					lines[i] = []byte(uniqueOtelComponentIDReplace.regexp.ReplaceAllString(stringLine, replacement))
+
+					// store the otel ID replaced without the space indentation and the colon to be replaced later
+					// (e.g. http_check/4391d954-1ffe-4014-a256-5eda78a71828 replaced by http_check/componentid-0)
+					otelID := strings.SplitN(strings.TrimSpace(stringLine), ":", 2)[0]
+					replacementsDone[otelID] = strings.SplitN(strings.TrimSpace(string(lines[i])), ":", 2)[0]
```

Comment:
Nit. I think these `strings.SplitN` could be replaced by `strings.Cut`.
```suggestion
					otelID, _, _ := strings.Cut(strings.TrimSpace(stringLine), ":")
					replacementsDone[otelID], _, _ = strings.Cut(strings.TrimSpace(string(lines[i])), ":")
```